File: framework/navigation-dxa-module/src/main/java/com/sdl/webapp/navigation/model/NavigationItem.java
package com.sdl.webapp.navigation.model;

import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.RichText;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import com.sdl.webapp.common.api.model.entity.Image;
import com.sdl.webapp.common.api.model.entity.Link;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

/**
 * Navigation Item
 *
 * @author nic
 */
@SemanticEntity(entityName = "NavigationItem", vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class NavigationItem extends AbstractEntityModel {

    @SemanticProperty("e:link")
    private Link link;

    @SemanticProperty("e:text")
    private RichText text;

    @SemanticProperty("e:image")
    private Image image;

    public Link getLink() {
        return link;
    }

    public RichText getText() {
        return text;
    }

    public Image getImage() {
        return image;
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.DXA.Modules.Navigation/Models/NavigationItem.cs
using Sdl.Web.Common.Models;

namespace SDL.DXA.Modules.Navigation.Models
{
    /// <summary>
    /// Navigation Item
    /// </summary>
    [SemanticEntity(Vocab = CoreVocabulary, EntityName = "NavigationItem", Prefix = "e")]
    public class NavigationItem : EntityModel
    {
        [SemanticProperty("e:link")]
        public Link Link { get; set; }

        [SemanticProperty("e:text")]
        public RichText Text { get; set; }

        [SemanticProperty("e:image")]
        public MediaItem Image { get; set; }
    }
}